### PR TITLE
Update GitHub Pages to kramdown [ci skip]

### PR DESCRIPTION
**Pull request by tammoippen**
GitHub changed the markdown processor from redcarped to kramdown, see [here](https://github.com/blog/2151-github-pages-drops-support-for-rdiscount-redcarpet-and-redcloth-textile-markup-engines). This PR applies the necessary changes and also adjust the syntax highlighting.

@jougs and @heplesser can you please review?

### CODE REVIEW

**Comment by heplesser** [comment]
@tammoippen I hope you didn't do this all by hand, but had a converter to help you!? What is the purpose of the new Gem file? Is there any way to preview the effect of your changes?

File: coding_guidelines_c++.md [hunk 1]
```diff
@@ -694,36 +714,38 @@ Copy as `.clang-format` file into directory, in which you execute the `clang-for
     SpacesInParentheses: true
     SpacesInSquareBrackets: true
     Standard: Cpp03
-    
+
     UseTab: false
```
Comment (jougs) [inline coding_guidelines_c++.md]: Couldn't this block me formatted as anonymous code using just a three-backticks-block without a language specification?

File: development_workflow.md [hunk 2]
```diff
@@ -446,61 +456,67 @@ collaborator.
...
-     git commit -am 'ENH - much better code'
-     git push origin master # pushes directly into your repo
+```bash
+git commit -am 'ENH - much better code'
+git push origin master # pushes directly into your repo
+```
 
 ----------------------------------------------------------------------------
 
-Use cases for the new platform
```
Comment (jougs) [inline development_workflow.md]: I think we should remove these use cases altogether. They were useful to convince everyone that the new platform can do everything we need, but now the workflow descriptions are much better suited for real work. Moreover, we never managed to formulate all of them into proper sentences.

File: stylesheets/pygment.css [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/* Based on from https://github.com/richleland/pygments-css/blob/master/zenburn.css*/
```
Comment (jougs) [inline stylesheets/pygment.css]: "on from" looks incomplete to me.

**Comment by jougs** [comment]
Many thanks! This looks good to me apart from the things in my comments. Could you please address those?

File: development_workflow.md (hunk 2 above)
Comment (tammoippen) [inline development_workflow.md]: Agree, but this should be a different PR.

**Comment by tammoippen** [comment]
@heplesser The `Gemfile` is for local testing. See this article on how to [test GitHub Pages locally](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/).

File: continuous_integration.md [hunk 4]
```diff
@@ -36,9 +36,9 @@ The current CI implementation is now available at the following URL:
 
     https://travis-ci.org/nest/nest-simulator
```
Comment (heplesser) [inline continuous_integration.md]: I think we should have this as a link, not as pre-formatted code.

File: continuous_integration.md [hunk 5]
```diff
@@ -36,9 +36,9 @@ The current CI implementation is now available at the following URL:
...
 
-Travis CI service has been integrated with the Github repository to automatically run the tests when code is pushed. Github integration is done by adding a simple YAML file to the project root.Travis results will appear in the Github pull requests and the primary log is visible in the Travis interface. 
+Travis CI service has been integrated with the Github repository to automatically run the tests when code is pushed. Github integration is done by adding a simple YAML file to the project root.Travis results will appear in the Github pull requests and the primary log is visible in the Travis interface.
 
-Build jobs
+## Build jobs
 
 The CI system is closely integrated to Github repository. Whenever some changes in the code
 are detected, the latest source code is downloaded to a worker machine which is running on Ubuntu 14.04 LTS(Trusty) and the following actions are performed:
```
Comment (heplesser) [inline continuous_integration.md]: The list of steps that  follows should be an enumerated list, not preformatted code.

File: development_workflow.md [hunk 6]
```diff
@@ -149,15 +156,15 @@ You could also use `pull`, which combines `fetch` and `merge`, as follows:
...
 Finally create a new branch for your work and check it out:
 
     git checkout -b my-new-feature master
 
-### Editing workflow
-#### Overview
+### Editing workflow overview
+
 ```bash
 # hack hack
```
Comment (heplesser) [inline development_workflow.md]: Why hack hack?

File: development_workflow.md [hunk 7]
```diff
@@ -636,25 +660,21 @@ If Abigail is an owner of the NEST repository, she can also add the label “doc
 
 If no one objects, the release manager merges the Pull Request into the public version of NEST.
 
-Alex wants to fix a bug in the NEST kernel
-------------------------------------------
+## Alex wants to fix a bug in the NEST kernel
```
Comment (heplesser) [inline development_workflow.md]: The workflows from here on down are unfinished work. I think we should remove them.

File: coding_guidelines_c++.md [hunk 8]
```diff
@@ -12,27 +12,8 @@ In the code review process we want to enforce a consistent coding style to impro
```
Comment (heplesser) [inline coding_guidelines_c++.md]: This file had `_c++` in its name and links to it point to C++ code guidelines, but the header just says "NEST Coding Style Guidelines". I think the header should mention C++, and we should have links to our other guides (SLI, Python). We might also add a section on Python here.

**Comment by heplesser** [comment]
@tammoippen I have tested the pages with Jekyll now and they largely look fine. :+1: once my items above are addressed.

**Comment by heplesser** [comment]
@tammoippen Running Jekyll generates a lot of files in the gh-pages directory, which certainly should not be checked in, and it modifies config.yml. What would be a good `.gitignore` strategy here?

File: development_workflow.md (hunk 6 above)
Comment (tammoippen) [inline development_workflow.md]: This is still from the initial version. I stated it differently.

File: development_workflow.md (hunk 7 above)
Comment (tammoippen) [inline development_workflow.md]: Same as above:

> Agree, but this should be a different PR.

**Comment by tammoippen** [comment]
@heplesser I only get the `_site/` directory and the `Gemfile.lock` generated and both are ignored by the `.gitignore`.  What other files do you mean? BTW: my `config.yml` is never changed by Jekyll.

**Comment by heplesser** [comment]
@tammoippen This is what it looks like on my machine. The files and directories are all at the top level of the working directory.

screen shot 2016-05-30 at 10 31 19

**Comment by tammoippen** [comment]
@heplesser I think you did the `Step 1` `git init my-jekyll-site-project-name` or the `Step 3` `bundle exec jekyll new . --force` from [test GitHub Pages locally](https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/). From that site, we only need the parts:

* Requirements (once)
* Step 2: Install Jekyll using Bundler (once)
* Step 4: Build your local Jekyll site
* Keeping your site up to date with the GitHub Pages gem

Sorry that I was not clearer on this.

**Comment by jougs** [comment]
:+1: and merging!
